mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt

```diff
@@ -11,42 +13,55 @@ import ch.epfl.sdp.mobile.ui.social.SocialScreen
 import ch.epfl.sdp.mobile.ui.social.SocialScreenState
 import ch.epfl.sdp.mobile.ui.social.SocialScreenState.Mode.Following
 import ch.epfl.sdp.mobile.ui.social.SocialScreenState.Mode.Searching
+import kotlinx.coroutines.flow.*
 
-private class SnapshotSocialScreenState(
-    private val following: State<List<Profile>>,
-) : SocialScreenState {
-
-  private data class ProfileAdapter(
-      private val profile: Profile,
-  ) : Person {
-    override val backgroundColor: Color
-      get() = profile.backgroundColor
-    override val name: String
-      get() = profile.name
-    override val emoji: String
-      get() = profile.emoji
-  }
-
-  override var mode: SocialScreenState.Mode by mutableStateOf(Following)
-
-  override var players: List<Person> = following.value.map { ProfileAdapter(it) }
-  override var input: String by mutableStateOf("")
-
-  // TODO :  define how to return to the following screen, return button ?
-  // TODO :  Modify here to update the player list when we do a search
-  override fun onValueChange() {
-    if (mode != Following) {
-      mode = Searching
-    }
-  }
+private data class ProfileAdapter(
+    private val profile: Profile,
+) : Person {
+  override val backgroundColor: Color
+    get() = profile.backgroundColor
+  override val name: String
+    get() = profile.name
+  override val emoji: String
+    get() = profile.emoji
 }
 
 @Composable
 fun StatefulFollowingScreen(
     user: AuthenticatedUser,
     modifier: Modifier = Modifier,
 ) {
-  val state = remember(user) { user.following }.collectAsState(emptyList())
+  val following =
+      remember(user) { user.following }.collectAsState(emptyList()).value.map { ProfileAdapter(it) }
```

Comment:
Nitpick : I'd suggest writing things using a property delegate, and mapping within the `remember {}` block. This way, you don't have to explicitly call `.value` on the `State`.

```kotlin
val following by remember(user) { user.following.map(::ProfileAdapter) }
  .collectAsState(emptyList())
```